stm32 xspi reference  to replace the ospi for stm32x mcu

FRASTM opened the pull request:
This change is for preparing the future stm32 series that could have spi/quad/octo/hexa spi peripheral and gives more flexibility to stm32 devices

The OSPI_xxx constant are renamed with XSPI_ prefix

The XSPI constants are common values to the existing zephyr  flash_stm32_qspi or flash_stm32_ospi.

requires https://github.com/zephyrproject-rtos/zephyr/pull/70434

FRASTM committed:
drivers: flash: stm32 ospi flash driver with common xspi constants

Adapt the stm32 octo flash driver to the XSPI model
and converting names from OSPI to OCTO
This is done through a xspi.h definition file

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: arm: st: stm32 devices with octospi includes common xspi header

change the header file to xspi. for the stm32 devices with octospi
peripheral.
Keep the flash_controller/ospi.h for bindings for compatibilty

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: st: stm32 boards with octospi instance include de xspi header

Use the xspi.h header file to configure the octospi node
of the stm32 disco kits

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebased on aeea1d10de8859d9368cbf07be0b9c3726c79a67

FRASTM commented:
CI failure due to https://github.com/zephyrproject-rtos/zephyr/issues/70445

erwango commented on include/zephyr/dt-bindings/flash_controller/ospi.h:
Don't remove purely, but add a deprecation notice + update migration guide

FRASTM replied:
updating the doc/releases/migration-guide-3.7 to introduce the new XSPI__ prefix

erwango commented on dts/arm/st/h5/stm32h562.dtsi:
```diff
@@ -5,7 +5,7 @@
 
 #include <st/h5/stm32h5.dtsi>
-#include <zephyr/dt-bindings/flash_controller/ospi.h>
+#include <zephyr/dt-bindings/flash_controller/xspi.h>
```
Keep both during the deprecation period ?

FRASTM committed:
doc: release: migration guide to stm32 XSPI constants

Update migration guide to introduce the XSPI_ generic
values for stm32 devices enabling the octoSPI.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
Closing this PR and change to XSPI for the stm32h5 serie only : see PR #70371
